### PR TITLE
feat: verify test vectors generated with both backends

### LINKED ISSUE #161: Test compatibility between paired and blstrs
@cryptonemo @dignifiedquire

### PR DESCRIPTION
**Pull request by cryptonemo**
Resolves #161 (or at least aims to)

### CODE REVIEW

**Review by dignifiedquire — APPROVED** [review]

**Comment by cryptonemo** [comment]
@nikkolasg please merge at will

File: src/groth16/aggregate/prove.rs [hunk 1]
```diff
@@ -400,18 +400,18 @@ pub(super) fn polynomial_evaluation_product_form_from_transcript<F: Field>(
...
+// challenges:
+//
+//     We wish to compute $(1+x_1ra)(1+x_0(ra)^2) = 1 +  x_1ra + x_0(ra)^2 + x_0x_1(ra)^3$
+//     Algorithm: $c_{-1} = [1]$; $c_j = c_{i-1} \| (x_{l-j} * c_{i-1})$; $r = r*r$
+//     $c_0 = c_{-1} \| (x_1 * r * c_{-1}) = [1] \| [rx_1] = [1, rx_1]$, $r = r^2$
+//     $c_1 = c_0 \| (x_0 * r^2c_0) = [1, rx_1] \| [x_0r^2, x_0x_1r^3] = [1, x_1r, x_0r^2, x_0x_1r^3]$
+//     which is equivalent to $f(a) = 1 + x_1ra + x_0(ra)^2 + x_0x_1r^2a^3$
+//
+// This method expects the coefficients in reverse order so transcript[i] =
+// x_{l-j}.
```
Comment (nikkolasg) [inline src/groth16/aggregate/prove.rs:414]: Why removing docs ? I thought these comments would be parsed as docs where there are 3 "/"

**Review by nikkolasg — APPROVED** [review]

File: src/groth16/aggregate/prove.rs (hunk 1 above)
Comment (cryptonemo) [reply src/groth16/aggregate/prove.rs:414]: The compiler tries to compile them apparently, and it causes build errors :-(

Comment (vmx) [reply src/groth16/aggregate/prove.rs:414]: It's the doc tests. Code blocks are expected to be Rust by default. You can either make it not a code block (by not indenting that part) or you could put it into a code fence and specify a type that is not rust, e.g. `text` or `latex`.

**Pull request merged**